presentation/feedprofile/src/main/java/com/hilingual/presentation/feedprofile/profile/FeedProfileViewModel.kt

```diff
@@ -48,23 +48,31 @@ internal class FeedProfileViewModel @Inject constructor(
         viewModelScope.launch {
             val feedProfileDeferred = async { feedRepository.getFeedProfile(targetUserId) }
             val sharedDiariesDeferred = async { feedRepository.getSharedDiaries(targetUserId) }
-            val likedDiariesDeferred = async { feedRepository.getLikedDiaries(targetUserId) }
 
             val feedProfileResult = feedProfileDeferred.await()
             val sharedDiariesResult = sharedDiariesDeferred.await()
-            val likedDiariesResult = likedDiariesDeferred.await()
 
-            if (feedProfileResult.isFailure || sharedDiariesResult.isFailure || likedDiariesResult.isFailure) {
+            if (feedProfileResult.isFailure || sharedDiariesResult.isFailure) {
                 feedProfileResult.onLogFailure {}
                 sharedDiariesResult.onLogFailure {}
-                likedDiariesResult.onLogFailure {}
                 _sideEffect.emit(FeedProfileSideEffect.ShowRetryDialog { loadFeedProfile() })
                 return@launch
             }
 
             val feedProfileInfoModel = feedProfileResult.getOrThrow()
             val sharedDiariesModel = sharedDiariesResult.getOrThrow()
-            val likedDiariesModel = likedDiariesResult.getOrThrow()
+
+            val likedDiariesModel = if (feedProfileInfoModel.isMine) {
+                val likedDiariesResult = feedRepository.getLikedDiaries(targetUserId)
+                if (likedDiariesResult.isFailure) {
+                    likedDiariesResult.onLogFailure {}
+                    _sideEffect.emit(FeedProfileSideEffect.ShowRetryDialog { loadFeedProfile() })
+                    return@launch
+                }
+                likedDiariesResult.getOrThrow()
+            } else {
+                null
+            }
```

Comment:
```suggestion
            val likedDiariesModel = if (feedProfileInfoModel.isMine) {
                feedRepository.getLikedDiaries(targetUserId)
                    .fold(
                        onSuccess = { it.diaryList },
                        onFailure = { throwable ->
                            Timber.e(throwable)
                            _sideEffect.emit(FeedProfileSideEffect.ShowRetryDialog { loadFeedProfile() })
                            return@launch
                        }
                    )
            } else {
                emptyList()
            }
```
onLogFailure가 결국 Result.onFailure인데 에러처리를 밖에서 해주면서 오로지 로그만을 위해 onLogFailure를 호출하는게 좋아보이진 않아요 fold를 사용해서 코드 정리해보면 어떨까요?? 제 생각에는 이것도 똑같이 동작할 것 같은뎅 

그리고 isMine이 아닐 때는 emptyList를 반환하는 것도 좋을 것 같아요! 이러면 밑에서 null 처리 안해줘도 될 것 같음